python-lib/gluonts_forecasts/model_handler.py

```diff
@@ -127,71 +40,57 @@ class ModelParameterError(ValueError):
-
-    Attributes:
-        model_name (str)
     """
 
-    def __init__(self, model_name):
+    def __init__(self, model_name, label, estimator_class=None, predictor_class=None, trainer_class=None):
         self.model_name = model_name
-        self.model_descriptor = self._get_model_descriptor()
-
-    def _get_model_descriptor(self):
-        model_descriptor = MODEL_DESCRIPTORS.get(self.model_name)
-        if model_descriptor is None:
-            return {}
-        else:
-            return model_descriptor
+        self.label = label
+        self.estimator_class = estimator_class
+        self.predictor_class = predictor_class
+        self.trainer_class = trainer_class
 
     def estimator(self, model_parameters, **kwargs):
-        default_kwargs = self.model_descriptor.get(DEFAULT_KWARGS, {})
-        kwargs.update(default_kwargs)
         model_parameters_sanitized = sanitize_model_parameters(model_parameters.get("kwargs", {}), self.model_name)
         kwargs.update(model_parameters_sanitized)
-        estimator = self.model_descriptor.get(ESTIMATOR)
         kwargs = self._convert_parameters_to_class(kwargs)
         try:
-            ret = None if estimator is None else estimator(**kwargs)
+            ret = None if self.estimator_class is None else self.estimator_class(**kwargs)
         except Exception as err:
-            raise ModelParameterError(
-                f"Issue with parameters '{kwargs}' of model '{self.model_name}'. Full error: {err}"
-            )
+            raise ModelParameterError(f"Issue with parameters '{kwargs}' of model '{self.label}'. Full error: {err}")
         return ret
 
     def trainer(self, **kwargs):
-        trainer = self.model_descriptor.get(TRAINER)
         try:
-            ret = None if trainer is None else trainer(**kwargs)
+            ret = None if self.trainer_class is None else self.trainer_class(**kwargs)
         except Exception as err:
             raise ModelParameterError(
-                f"Issue with parameters '{kwargs}' of model trainer for '{self.model_name}'. Full error: {err}"
+                f"Issue with parameters '{kwargs}' of model trainer for '{self.label}'. Full error: {err}"
             )
         return ret
 
     def predictor(self, **kwargs):
-        predictor = self.model_descriptor.get(PREDICTOR)
         try:
-            ret = None if predictor is None else predictor(**kwargs)
+            ret = None if self.predictor_class is None else self.predictor_class(**kwargs)
         except Exception as err:
             raise ModelParameterError(
-                f"Issue with parameters '{kwargs}' of model predictor for {self.model_name}. Full error: {err}"
+                f"Issue with parameters '{kwargs}' of model predictor for {self.label}. Full error: {err}"
             )
         return ret
 
     def can_use_batch_size(self):
-        return self.model_descriptor.get(CAN_USE_BATCH_SIZE, False)
+        return False
```

Comment:
Not sure it is a good idea to let these be defined. It is not necessarily desirable to have the base class define default behaviors for children classes. If you want developers writing children classes to state explicitly all of these methods, then it would be better to:
```suggestion
        raise NotImplementedError
```
Happy to elaborate more to see what's the right choice.

Reply:
For me, a model has no specific capabilities by default (so everything to False) and then it can overwrite some methods if it has some specific capabilities. I got inspired by how dip handles all the SQL dialects, the parent class implements a default behavior and the children (the different SQL engines) overwrite some methods if they have different behaviors.

Reply:
That's a reasonable assumption. I think it is worth explaining this in the docstring of the base class, as this is not completely trivial.

Reply:
14bc54b